lib/src/revset.rs

```diff
@@ -223,6 +223,7 @@ pub enum RevsetExpression {
         consequent: Rc<Self>,
         alternate: Rc<Self>,
     },
+    Coalesce(Vec<Rc<Self>>),
```

Comment:
I would transform `coalesce(x, y, z..)` to binary form `Coalesce(x, Coalesce(y, z))` because it's easier to process in later pass (and the chain wouldn't be that long.)